contracts/interfaces/IEnum.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
```

Comment:
Is there a reason to add `Enum.sol` here again as an interface?
There is an abstract instance of `Enum.sol` in `common/Enum.sol`. 
I would suggest to keep only one instance from either of them.

Reply:
An interface can only inherit from another interface, so I had to create a new interface for the Enum.

Reply:
Could we utilize the enum interface everywhere instead of the other one?

Reply:
We can 👍🏾 Updated in the latest commit.